### PR TITLE
feat: Implement HTTP Service

### PR DESCRIPTION
**Pull request by dipasqualew**
The HTTP Service will:

1. Wrap axios and perform axios requests via axios.request
2. Enforce a standard timeout of 10s
3. Check whether the `x-comicrelief-test-metadata` header is set, in which case the header will be added to downstream requests

See:
- [ENG-363]

[ENG-363]: https://comicrelief.atlassian.net/browse/ENG-363

### CODE REVIEW

**Review by seb-cr — COMMENTED** [review]
Good work! I've left a couple of minor comments, as always.

File: src/Service/HTTP.service.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+   * @param config
+   */
+  async request(config: AxiosRequestConfig): Promise<AxiosResponse> {
+    const mergedConfig = {
+      timeout: this.config.timeout,
+      headers: {},
+      ...config,
+    };
+
+    const event = this.getContainer().get(this.definitions.REQUEST);
```
[the remark this answers is not in the document]
Comment (dipasqualew) [reply src/Service/HTTP.service.js:41]: That is true, but in the context of Axios having `request` is confusing, this is why I opted for `event`

File: tests/unit/Service/HTTP.service.test.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      'GET Request': { method: 'GET', url: '/' },
+      'POST Request': { method: 'POST', url: '/' },
+      'PUT Request': { method: 'PUT', url: '/' },
+      'PATCH Request': { method: 'PATCH', url: '/' },
+      'HEAD Request': { method: 'HEAD', url: '/' },
+      'DELETE Request': { method: 'DELETE', url: '/' },
+      'with URL': { url: '/some/nested/path' },
+      'with baseURL': { baseUrl: 'https://comicrelief.com/test', url: '/additional/url' },
+      'overriding timeout': { timeout: 99 },
+      'with headers': { headers: { Authorization: 'Bearer test' } },
```
[the remark this answers is not in the document]
Comment (dipasqualew) [reply tests/unit/Service/HTTP.service.test.js:27]: Good catch! 95f4e440c76d82b0042be38d226989b3ab3377e9

File: src/Service/HTTP.service.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (seb-cr) [reply src/Service/HTTP.service.js:41]: Ooh, fair point, I somehow missed that the method name was also `request` 😄 

Maybe `lambdaRequest` or `requestService` then? Or avoid by inlining? Or keep `event` if you think it's unambiguous.

[the remark this answers is not in the document]
Comment (dipasqualew) [reply src/Service/HTTP.service.js]: I am happy with lambdaRequest! 0a023f005e1c90689aa8491b51c102a80a627fa5

[the remark this answers is not in the document]
Comment (seb-cr) [reply src/Service/HTTP.service.js]: Cool, thanks 🙂

**Review by seb-cr — APPROVED** [review]

**Pull request merged**